api/spec/src/oauth.tsp

```diff
@@ -0,0 +1,130 @@
+import "@typespec/http";
+
+namespace OAuth2;
```

Comment:
I wonder if this is the best place for this, especially because it directly says "client" and "app" below. Shouldn't it just be in the apps package?